tests/Greenspun/REPL.cpp

```diff
@@ -0,0 +1,167 @@
+
+#include <iostream>
+
+#include "Greenspun/EvalResult.h"
+#include "Greenspun/Interpreter.h"
+#include "Greenspun/Primitives.h"
+#include "velocypack/Builder.h"
+#include "velocypack/Parser.h"
+#include "velocypack/velocypack-aliases.h"
+
+#include "Utilities/Completer.h"
+#include "Utilities/LineEditor.h"
+#include "Utilities/ShellBase.h"
+
+/* YOLO */
+
+using namespace arangodb;
+
+greenspun::EvalResult Func_thisId(greenspun::Machine& ctx,
+                                  VPackSlice const params, VPackBuilder& result) {
+  result.add(VPackValue("V/1"));
+  return {};
+}
+
+std::unordered_map<std::string, VPackBuilder> variableValues;
+
+greenspun::EvalResult Func_VarSet(greenspun::Machine& ctx,
+                                  VPackSlice const params, VPackBuilder& result) {
+  if (params.length() != 2) {
+    return greenspun::EvalError("expected two parameters");
+  }
+  VPackSlice name = params.at(0);
+  VPackSlice value = params.at(1);
+  if (!name.isString()) {
+    return greenspun::EvalError("expected string as first parameter, found: " +
+                                name.toJson());
+  }
+
+  auto nameStr = name.copyString();
+  VPackBuilder& builder = variableValues[nameStr];
+  builder.clear();
+  builder.add(value);
+
+  ctx.setVariable(nameStr, builder.slice());
+  return {};
+}
+
+void AddSomeFunctions(greenspun::Machine& m) {
+  m.setFunction("this-id", Func_thisId);
+  m.setFunction("var-set!", Func_VarSet);
+}
+
+struct LispCompleter : arangodb::Completer {
+  bool isComplete(std::string const& source, size_t /*lineno*/) final {
+    int openBrackets = 0;
+    int openBraces = 0;
+    int openStrings = 0;  // only used for template strings, which can be multi-line
+    int openComments = 0;
```

Comment:
```suggestion
    int openBrackets = 0;
    int openBraces = 0;
```
I think these two aren't used anymore, either.

Reply:
Done. (Also needed to remove them before the return in the comparison statement.)